sky/clouds/service_catalog/images/provisioners/skypilot.sh

```diff
@@ -62,26 +63,29 @@ conda activate skypilot-runtime
 $([ -s ~/.sky/python_path ] && cat ~/.sky/python_path 2> /dev/null || which python3) -m venv ~/skypilot-runtime --system-site-packages
 echo "$(echo ~/skypilot-runtime)/bin/python" > ~/.sky/python_path
 source ~/.bashrc
+PYTHON_PATH=$([ -s ~/.sky/python_path ] && cat ~/.sky/python_path 2> /dev/null || which python3)
```

Comment:
`PYTHONPATH` is a reserved env var by python. Altough we use `PYTHON_PATH` with a `_` here, it might be better to avoid the potential confusion. Let's rename this var to `PYTHON_EXEC`?